lxd/network/network_utils.go

```diff
@@ -1164,3 +1164,16 @@ func InterfaceBindWait(ifName string) error {
 
 	return fmt.Errorf("Bind of interface %q took too long", ifName)
 }
+
+// IPRangesOverlap checks whether two ip ranges have ip addresses in common
+func IPRangesOverlap(r1, r2 *shared.IPRange) bool {
```

Comment:
It would be good to add some unit tests for this function, see examples in https://github.com/lxc/lxd/blob/master/lxd/network/network_utils_test.go